admin/src/components/Title.js

```diff
@@ -1,61 +1,101 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import Container from "@mui/material/Container";
 import { useSnackbar } from "notistack";
 import { errorHandlerWithSnackbar } from "../errors";
-import { TableCell, TableRow, TextField } from "@mui/material";
+import { Autocomplete, TextField, Button,
+    TableCell, TableRow,
+    Dialog, DialogTitle, DialogContent, DialogActions,
+    CircularProgress, Card, Stack } from "@mui/material";
 import ShowPresetButton from "./ShowPresetButton";
 import Box from "@mui/material/Box";
 import IconButton from "@mui/material/IconButton";
 import EditIcon from "@mui/icons-material/Edit";
 import SaveIcon from "@mui/icons-material/Save";
 import DeleteIcon from "@mui/icons-material/Delete";
-import { PresetsTableCell } from "./PresetsTableCell";
+import PreviewIcon from "@mui/icons-material/Preview";
+import { PresetsTableCell, ValueEditorPropTypes } from "./PresetsTableCell";
 import PropTypes from "prop-types";
 import { activeRowColor } from "../styles";
 import { onChangeFieldHandler } from "./PresetsTableRow";
 import { PresetsManager } from "./PresetsManager";
 import { usePresetWidgetService } from "../services/presetWidget";
+import { useTitleWidgetService } from "../services/titleWidget";
 
-const parseJSONOrDefault = (text, defult) => {
-    try {
-        return JSON.parse(text);
-    } catch (e) {
-        return defult;
-    }
+const PreviewSVGDialog = ({ open, handleClose, id }) => {
+    const { enqueueSnackbar } = useSnackbar();
+    const service = useTitleWidgetService("/title", errorHandlerWithSnackbar(enqueueSnackbar), false);
+    const [content, setContent] = useState(undefined);
+    useEffect(() => open && service.getPreview(id).then(r => setContent(r.content)), [open, id]);
+    return (
+        <Dialog onClose={handleClose} open={open} fullWidth maxWidth="md">
```

Comment:
Speaking in OOP terms, you are extending the MUI dialog to fit your specific purpose.
But it's still a variation of MUI Dialog.
That's encapsulation.
I would rather stick to the original prop names of MUI of "onClose".
And pass them with {...props} passing all the props that might be needed for this Dialog.